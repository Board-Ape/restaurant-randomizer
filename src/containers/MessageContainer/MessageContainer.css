h1 {
  font-size: 40px;
  background-color: black;
  color: #460C7F;
  margin: 0;
  padding-bottom: 20px; }
