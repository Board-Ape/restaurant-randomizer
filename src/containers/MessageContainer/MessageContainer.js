import React, { Component } from 'react';
import './MessageContainer.css';
import { connect } from 'react-redux';
import { addEventName } from '../../actions';
import MessageCard from '../../components/MessageCard/MessageCard';
import PropTypes from 'prop-types';

class MessageContainer extends Component {
  constructor() {
    super();
    this.state = {
      eventTitle: ''
    };
  }

  handleChange = (event) => {
    this.setState({eventTitle: event.target.value});
  }

  submitEventName = (event) => {
    event.preventDefault();
    this.props.handleSubmit(this.state.eventTitle, this.props.eventName.length);
  }

  renderMessageCards = () => {
    const messageCardArray = this.props.eventName.map( (message) => {
      return (<MessageCard
        key={message.index}
        eventName={message.name}
      />);
    });
    return messageCardArray;
  }

  render() {

    return (
      <div className='add-message-container'>
        <h2 className='message-container-title'>AddMessage!</h2>
        <form onSubmit={this.submitEventName}>
          <input
            className='text-box'
            type='text'
            placeholder='Text Message'
            value={this.state.eventTitle}
            onChange={this.handleChange}
          />
          <input className='submit-button' type='submit' />
        </form>
        <div>{this.renderMessageCards()}</div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return { eventName: state.eventName };
};

const mapDispatchToProps = (dispatch) => {
  return {
    handleSubmit: (eventTitle, id) => {
      dispatch(addEventName(eventTitle, id));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MessageContainer);

MessageContainer.propTypes = {
  handleSubmit: PropTypes.func,
  eventName: PropTypes.array
};
