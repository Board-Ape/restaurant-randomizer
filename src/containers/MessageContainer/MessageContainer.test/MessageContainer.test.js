import React from 'react';
import {MessageContainer, mapStateToProps, mapDispatchToProps} from '../MessageContainer';
import { shallow } from 'enzyme';

describe('MessageContainer tests', () => {
  let messageContainer;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      eventName: [],
      handleSubmit: jest.fn()
    };
    messageContainer = shallow(
      <MessageContainer {...mockProps}/>
    );
  });

  it('should render correctly', () => {
    expect(messageContainer).toBeDefined();
  });

  it('should mactch the snapshot', () => {
    expect(messageContainer).toMatchSnapshot();
  });

  it('should mount with the correct elements', () => {
    expect(messageContainer.find('div').length).toEqual(2);
    expect(messageContainer.find('h2').length).toEqual(1);
    expect(messageContainer.find('form').length).toEqual(1);
    expect(messageContainer.find('input').length).toEqual(2);
  });
});

describe('mapStateToProps tests', () => {
  let mockStore;
  let results;

  beforeEach(() => {
    mockStore = {
      eventName: ['Jorge', 'Jason', 'DJR', 'Sam']
    };
    results = mapStateToProps(mockStore);
  });

  it('should pull message data from the store', () => {
    expect(results.eventName).toEqual(mockStore.eventName);
  });
});

describe('mapDispatchToProps tests', () => {
  let mockDispatch;
  let results;

  beforeEach(() => {
    mockDispatch = jest.fn();
    results = mapDispatchToProps(mockDispatch);
  });

  it('should dispatch correctly when storing messages', () => {
    results.handleSubmit();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
