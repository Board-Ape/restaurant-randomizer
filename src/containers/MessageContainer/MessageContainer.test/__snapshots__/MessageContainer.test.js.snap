// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContainer tests should mactch the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageContainer
    eventName={Array []}
    handleSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="message-container-title"
        >
          AddMessage!
        </h2>,
        <form
          onSubmit={[Function]}
        >
          <input
            className="text-box"
            onChange={[Function]}
            placeholder="Text Message"
            type="text"
            value=""
          />
          <input
            className="submit-button"
            type="submit"
          />
        </form>,
        <div />,
      ],
      "className": "add-message-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "AddMessage!",
          "className": "message-container-title",
        },
        "ref": null,
        "rendered": "AddMessage!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="text-box"
              onChange={[Function]}
              placeholder="Text Message"
              type="text"
              value=""
            />,
            <input
              className="submit-button"
              type="submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "text-box",
              "onChange": [Function],
              "placeholder": "Text Message",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "submit-button",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="message-container-title"
          >
            AddMessage!
          </h2>,
          <form
            onSubmit={[Function]}
          >
            <input
              className="text-box"
              onChange={[Function]}
              placeholder="Text Message"
              type="text"
              value=""
            />
            <input
              className="submit-button"
              type="submit"
            />
          </form>,
          <div />,
        ],
        "className": "add-message-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "AddMessage!",
            "className": "message-container-title",
          },
          "ref": null,
          "rendered": "AddMessage!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="text-box"
                onChange={[Function]}
                placeholder="Text Message"
                type="text"
                value=""
              />,
              <input
                className="submit-button"
                type="submit"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "text-box",
                "onChange": [Function],
                "placeholder": "Text Message",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "submit-button",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
